### ISSUE TITLE
[Bug]:  integrate_with_hugging-face sample code is error

### ISSUE THREAD

**Issue by EvanZhou666** [milvus-io/milvus#31389]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone):
- MQ type(I don't know , I did't use any mq):    
- SDK version(e.g. pymilvus 2.3.7):
- OS(Milvus Server is Ubuntu, Client SDK runned on windows): 
- CPU/Memory: 2C4G
- GPU: No
- Others:
```

### Current Behavior

When I run the sample code , throw error:
Map: 100%|██████████| 99/99 [00:00<00:00, 2068.59 examples/s]
Map:   0%|          | 0/99 [00:00<?, ? examples/s]
Process finished with exit code -1073741819 (0xC0000005)

The Sample code is copied from [here](https://milvus.io/docs/integrate_with_hugging-face.md) 

`from pymilvus import connections, FieldSchema, CollectionSchema, DataType, Collection, utility
from datasets import load_dataset_builder, load_dataset, Dataset
from transformers import AutoTokenizer, AutoModel
from torch import clamp, sum

# 参数
DATASET = "squad"  # Huggingface Dataset to use
MODEL = 'bert-base-uncased'  # Transformer to use for embeddings
TOKENIZATION_BATCH_SIZE = 1000  # Batch size for tokenizing operation
INFERENCE_BATCH_SIZE = 64  # batch size for transformer
INSERT_RATIO = .001  # How many titles to embed and insert
COLLECTION_NAME = 'huggingface_db'  # Collection name
DIMENSION = 768  # Embeddings size
LIMIT = 10  # How many results to search for
MILVUS_HOST = "localhost"
MILVUS_PORT = "19530"

# create collection

connections.connect(host=MILVUS_HOST, port=MILVUS_PORT)

if utility.has_collection(COLLECTION_NAME):
    utility.drop_collection(COLLECTION_NAME)

fields = [
    FieldSchema(name='id', dtype=DataType.INT64, is_primary=True, auto_id=True),
    FieldSchema(name='original_question', dtype=DataType.VARCHAR, max_length=1000),
    FieldSchema(name='answer', dtype=DataType.VARCHAR, max_length=1000),
    FieldSchema(name='original_question_embedding', dtype=DataType.FLOAT_VECTOR, dim=DIMENSION)
]
schema = CollectionSchema(fields=fields)
collection = Collection(name=COLLECTION_NAME, schema=schema)

index_params = {
    'metric_type': 'L2',
    'index_type': "IVF_FLAT",
    'params': {"nlist": 1536}
}
collection.create_index(field_name="original_question_embedding", index_params=index_params)
collection.load()

# insert data

data_dataset = load_dataset(DATASET, split='all')
data_dataset = data_dataset.train_test_split(test_size=INSERT_RATIO, seed=42)['test']
data_dataset = data_dataset.map(lambda val: {'answer': val['answers']['text'][0]}, remove_columns=['answers'])

tokenizer = AutoTokenizer.from_pretrained(MODEL)

def tokenize_question(batch):
    results = tokenizer(batch['question'], add_special_tokens=True, truncation=True, padding="max_length",
                        return_attention_mask=True, return_tensors="pt")
    batch['input_ids'] = results['input_ids']
    batch['token_type_ids'] = results['token_type_ids']
    batch['attention_mask'] = results['attention_mask']
    return batch

data_dataset = data_dataset.map(tokenize_question, batch_size=TOKENIZATION_BATCH_SIZE, batched=True)
data_dataset.set_format('torch', columns=['input_ids', 'token_type_ids', 'attention_mask'], output_all_columns=True)

model = AutoModel.from_pretrained(MODEL)

def embed(batch):
    sentence_embs = model(
        input_ids=batch['input_ids'],
        token_type_ids=batch['token_type_ids'],
        attention_mask=batch['attention_mask']
    )[0]
    input_mask_expanded = batch['attention_mask'].unsqueeze(-1).expand(sentence_embs.size()).float()
    batch['question_embedding'] = sum(sentence_embs * input_mask_expanded, 1) / clamp(input_mask_expanded.sum(1),
                                                                                      min=1e-9)
    return batch

data_dataset = data_dataset.map(embed, remove_columns=['input_ids', 'token_type_ids', 'attention_mask'], batched=True,
                                batch_size=INFERENCE_BATCH_SIZE)

def insert_function(batch):
    insertable = [
        batch['question'],
        [x[:995] + '...' if len(x) > 999 else x for x in batch['answer']],
        batch['question_embedding'].tolist()
    ]
    collection.insert(insertable)

data_dataset.map(insert_function, batched=True, batch_size=64)
collection.flush()

# 提问
questions = {'question': ['When was chemistry invented?', 'When was Eisenhower born?']}
question_dataset = Dataset.from_dict(questions)

question_dataset = question_dataset.map(tokenize_question, batched=True, batch_size=TOKENIZATION_BATCH_SIZE)
question_dataset.set_format('torch', columns=['input_ids', 'token_type_ids', 'attention_mask'], output_all_columns=True)
question_dataset = question_dataset.map(embed, remove_columns=['input_ids', 'token_type_ids', 'attention_mask'],
                                        batched=True, batch_size=INFERENCE_BATCH_SIZE)

def search(batch):
    res = collection.search(batch['question_embedding'].tolist(), anns_field='original_question_embedding', param={},
                            output_fields=['answer', 'original_question'], limit=LIMIT)
    overall_id = []
    overall_distance = []
    overall_answer = []
    overall_original_question = []
    for hits in res:
        ids = []
        distance = []
        answer = []
        original_question = []
        for hit in hits:
            ids.append(hit.id)
            distance.append(hit.distance)
            answer.append(hit.entity.get('answer'))
            original_question.append(hit.entity.get('original_question'))
        overall_id.append(ids)
        overall_distance.append(distance)
        overall_answer.append(answer)
        overall_original_question.append(original_question)
    return {
        'id': overall_id,
        'distance': overall_distance,
        'answer': overall_answer,
        'original_question': overall_original_question
    }

question_dataset = question_dataset.map(search, batched=True, batch_size=1)
for x in question_dataset:
    print()
    print('Question:')
    print(x['question'])
    print('Answer, Distance, Original Question')
    for x in zip(x['answer'], x['distance'], x['original_question']):
        print(x)
`

### Expected Behavior

No errors  .   The Expected Behavior is same as [tutorial](https://milvus.io/docs/integrate_with_hugging-face.md) 

### Milvus Log

● milvus.service - Milvus Standalone Server
     Loaded: loaded (/lib/systemd/system/milvus.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2024-03-18 17:05:45 CST; 19h ago
   Main PID: 1083721 (milvus)
      Tasks: 26 (limit: 4363)
     Memory: 451.1M
     CGroup: /system.slice/milvus.service
             └─1083721 /usr/bin/milvus run standalone

Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.953 +08:00] [INFO] [querynodev2/services.go:1383] ["sync action"] [traceID=eb7fee9c307be1c11b2e8031e981>
Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.954 +08:00] [INFO] [delegator/distribution.go:294] ["Update readable segment version"] [oldVersion=1710>
Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.954 +08:00] [INFO] [datacoord/services.go:781] ["get recovery info request received"] [traceID=e93f20c1>
Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.954 +08:00] [INFO] [datacoord/handler.go:116] [GetQueryVChanPositions] [collectionID=448463568196428829>
Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.954 +08:00] [INFO] [datacoord/handler.go:299] ["channel seek position set from channel checkpoint meta">
Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.954 +08:00] [INFO] [datacoord/services.go:796] ["datacoord append channelInfo in GetRecoveryInfo"] [tra>
Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.954 +08:00] [INFO] [datacoord/services.go:781] ["get recovery info request received"] [traceID=90bd28b7>
Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.954 +08:00] [INFO] [datacoord/handler.go:116] [GetQueryVChanPositions] [collectionID=448461159922052884>
Mar 19 12:08:30 hcss-ecs-66b7 milvus[1083721]: [2024/03/19 12:08:30.954 +08:00] [INFO] [datacoord/handler.go:299] ["channel seek position set from channel checkpoint meta">

**Comment by yanliang567** [comment]
/assign @jaelgu 
could you please help to take a look here?
/unassign

**Comment by yanliang567** [comment]
@jaelgu any updates

**Comment by yanliang567** [comment]
@jaelgu any updates?

**Comment by zc277584121** [comment]
thanks for your feedback. currently the documentation is out of date, we will update the documents soon before May 30th @EvanZhou666

**Comment by jaelgu** [comment]
@EvanZhou666 Hello, we've updated the tutorial: https://milvus.io/docs/integrate_with_hugging-face.md

**Comment by jaelgu** [comment]
> Hello!
> 
> It is recommended to use Sentence Transformers to use these models, rather than Transformers directly, to avoid unexpected bugs with e.g. pooling modes. It would simplify the usage a lot, too. Would you consider updating the documentation?
> 
> * Tom Aarsen

Transformers offer more model options. If you want to use Sentence Transformers, we have another tutorial: https://milvus.io/docs/integrate_with_sentencetransformers.md

**Comment by tomaarsen** [comment]
You're very right, apologies, I missed that page. Looks great!

- Tom Aarsen

**Comment by yanliang567** [comment]
close as comments above
